prereise/gather/demanddata/bldg_electrification/load_projection.py

```diff
@@ -0,0 +1,674 @@
+        return other.stats[f"frac_ff_{clas}_heat"] - self.stats[f"frac_ff_{clas}_heat"]
+
+    def frac_dhw_ff2hp(self, other, clas):
+        # fraction of floor area electrified for dhw
+        return other.stats[f"frac_ff_dhw_{clas}"] - self.stats[f"frac_ff_dhw_{clas}"]
+
+    def frac_cook_ff2hp(self, other, clas):
+        # fraction of floor area electrified for cooking
+        cook_other = "cook" if clas == "com" else "other"
+        return (
+            other.stats[f"frac_ff_{cook_other}_{clas}"]
+            - self.stats[f"frac_ff_{cook_other}_{clas}"]
+        )
+
+    def frac_cooling_eff_change(self, other):
+        # account for future cooling efficiency improve
+        return self.cool_energy_intensity / other.cool_energy_intensity
+
+    def compare_hp_heat_type(self, other):
+        # return True if the heat pump type for projection scenario is the same as that of base scenario
+        return self.hp_type_heat == other.hp_type_heat
+
+
+def temp_to_energy(temp_series, hourly_fits_df, db_wb_fit, base_scen, hp_heat_cop):
+    """Compute baseload, heating, and cooling electricity for a certain hour of year under model base year scenario
+
+    :param pandas.Series temp_series: data for the given hour.
+    :param pandas.DataFrame hourly_fits_df: hourly and week/weekend breakpoints and
+        coefficients for electricity use equations.
+    :param class scenarios base_scen: reference scenario class
+    :return: (*list*) -- [baseload, heating, cooling]
+    """
+    temp = temp_series["temp_c"]
+    temp_wb = temp_series["temp_c_wb"]
+    dark_frac = temp_series["hourly_dark_frac"]
+    zone_hour = temp_series["hour_local"]
+
+    heat_eng = 0
+    mid_cool_eng = 0
+    cool_eng = 0
+    hp_eng = 0
+    resist_eng = 0
+
+    wk_wknd = (
+        "wk"
+        if temp_series["weekday"] < 5 and ~temp_series["holiday"]  # boolean value
+        else "wknd"
+    )
+
+    (
+        t_bpc,
+        t_bph,
+        i_heat,
+        s_heat,
+        s_dark,
+        i_cool,
+        s_cool_db,
+        s_cool_wb,
+    ) = (
+        hourly_fits_df.at[zone_hour, f"t.bpc.{wk_wknd}.c"],
+        hourly_fits_df.at[zone_hour, f"t.bph.{wk_wknd}.c"],
+        hourly_fits_df.at[zone_hour, f"i.heat.{wk_wknd}"],
+        hourly_fits_df.at[zone_hour, f"s.heat.{wk_wknd}"],
+        hourly_fits_df.at[zone_hour, f"s.dark.{wk_wknd}"],
+        hourly_fits_df.at[zone_hour, f"i.cool.{wk_wknd}"],
+        hourly_fits_df.at[zone_hour, f"s.cool.{wk_wknd}.db"],
+        hourly_fits_df.at[zone_hour, f"s.cool.{wk_wknd}.wb"],
+    )
+
+    base_eng = s_heat * t_bph + s_dark * dark_frac + i_heat
+
+    if temp <= t_bph:
+        heat_eng = -s_heat * (t_bph - temp)
+        # seperate resistance heat and heat pump energy by COP. HP assumed to be midium performance HP that's used in Mike's Joule paper
+        # ! Since we now have our estimation of current heat pump adoption rate, this function can be merged to the same one in zone_profile_generator.py
```

Comment:
Does it mean we will merge/delete this function after the logic of heat pump penetration model is merged?

Reply:
We can, I'm considering it. The only drawback of merging this function in zone_profile_generator.py is that we have to introduce more assumptions to the a pure linear regression script. The accuracy of the heat pump penetration model and the COPs we assumption is much less accurate compares to the load-temperature regression model.

Reply:
I see. Let's leave it as it is for now then.